### ISSUE TITLE
Accessibility features 

### ISSUE THREAD

**Issue by jbrgfx** [mdgriffith/stylish-elephants#11]
Something like http://package.elm-lang.org/packages/tesk9/accessible-html/latest would be a nice addition.  Accessibility is never achieved as an afterthought; it is too difficult so it needs to be build in from the start.

Your library is a fantastic improvement over CSS as it entirely removes one bothersome pain point from the stack, but I cannot see how to achieve simple accessibility features like role or rel or title like I can using  tesk9/accessible-html -- that, however, requires me to go back to working with the DOM and CSS.  Your decorative image, for example, fails to meet the WAI spec. -- it needs an empty ATL tag.

Doing things the ELM way is so very assuring, and having that for styles is really great, except when the core libraries do support WAI and one cannot incorporate those that do cause they are so opinionated about DOM and CSS.  Mixing the two sides is possible, but it makes the ELM code impossible to explain.

**Comment by mdgriffith** [comment]
Hello!

First, thanks, you're right, accessibility should be built in from the start.

Tessa's library is fantastic and has been an inspiration for a while.

This library already has quite a few accessibility concerns directly built in.

* `decorativeImage` does have an empty `alt` tag 😄 
* Labels for inputs are mandatory.
* general accessibility markup like `role=navigation` and `<h1>` tags have been moved to the `Element.Area` module.
* The `description` attribute is a general way to add a description to an element that will be picked up by screen readers.

I'm happy to improve accessibility of the library, so if you have specifics about something that's lacking, let me know and I'd be happy to look into it.

**Comment by jbrgfx** [comment]
Thanks for the pointer on Element.Area.role.  I will read 5.x  I will also try to see what I missed in implementing a decorativeImage (maybe you meant in 5.x ?

**Comment by jbrgfx** [comment]
Yes, Element.Area does look like it may fill some gaps.
Thanks.

**Comment by jbrgfx** [comment]
Ok, Element in 5.x does provide exactly what I was looking for:

row []
        <|
            [ image
                [ width (px 180)
                , height (px 70)
                ]
                { description = "elm logo", src = "elm_logo.png" }
            ]
{- output 
   <img class=" se el width-exact width-px-180 height-px-70" src="elm_logo.png" alt="elm logo">
-}
And:

row []
        <|
            [ decorativeImage
                [ width (px 180)
                , height (px 70)
                ]
                { src = "elm_logo.png" }
            ]
{- output:
<img class=" se el width-exact width-px-180 height-px-70" src="elm_logo.png" alt="">
-}
Excellent and thank you for great work.
